components/Village/Add/index.vue

```diff
@@ -0,0 +1,419 @@
+     * make modal visible or not
+     */
+    show: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      modalShow: false,
+      loading: false,
+      form: {
+        id: '',
+        name: '',
+        city_id: '',
+        district_id: '',
+        level: 0,
+        longitude: '',
+        latitude: ''
+      },
+      errors: {
+        villageId: null,
+        villageName: null,
+        villageLevel: null,
+        city: null,
+        district: null,
+        longitude: null,
+        latitude: null
+      },
+      listCity: [],
+      listDistrict: [],
+      optionsLevel: [
+        {
+          value: null,
+          label: 'Belum ada level'
+        },
+        {
+          value: 1,
+          label: 1
+        },
+        {
+          value: 2,
+          label: 2
+        },
+        {
+          value: 3,
+          label: 3
+        },
+        {
+          value: 4,
+          label: 4
+        }
+      ],
+      isDisabledOptionDistricts: true
+    }
+  },
+  computed: {
+    optionsCity () {
+      let city = []
+      if (Array.isArray(this.listCity) && this.listCity.length) {
+        city = this.listCity.map((item) => {
+          return {
+            value: item.id,
+            label: item.name
+          }
+        })
+      }
+      return city
+    },
+    optionsDistrict () {
+      let districts = []
+      if (Array.isArray(this.listDistrict) && this.listDistrict.length) {
+        districts = this.listDistrict.map((item) => {
+          return {
+            value: item.id,
+            label: item.name
+          }
+        })
+      }
+      return districts
+    },
+    isFormCompleted () {
+      return !!((
+        this.form.id.length &&
+        this.form.name.length &&
+        (this.form.city_id === undefined || this.form.city_id.length) &&
+        (this.form.district_id === undefined || this.form.district_id.length) &&
+        !(this.form.level === 0 || this.form.level === undefined) &&
+        this.form.longitude.length &&
+        this.form.latitude.length &&
+        !this.errors.villageId &&
+        !this.errors.villageName &&
+        !this.errors.city &&
+        !this.errors.district &&
+        !this.errors.longitude &&
+        !this.errors.latitude
+      ))
+    }
+  },
+  watch: {
+    show: {
+      handler (val) {
+        this.modalShow = val
+      },
+      immediate: true
+    },
+    'form.id' () {
+      // This pattern for checking user only allow input number and character (.)
+      const formatText = /(?=.*[^\d.])/g
+      if (formatText.test(this.form.id)) {
+        this.form.id = ''
+      }
+      // formatId is pattern for id village maximal 13 chars (10 digit and 3 symbol) => XX.XX.XX.XXXX
+      const formatId = /(\B(?=(\d{2})+(?!\d))(?=.{4}))/g
+      const newId = this.form.id.replaceAll(/[^0-9]/g, '').replace(formatId, '.')
+      this.form.id = newId
+      if (this.form.id.length < 13 && this.form.id.length > 13) {
```

Comment:
Ask:

Hello @doohanas If the `form.id.length` must be exactly 13 characters, wouldn't it easier to just write
```js
if (this.form.id.length !=== 13) {
  // ...
}
```

🤔

Reply:
Hello @Ibwedagama, I see it's more readable than I wrote 😆.
But I will refactor it again into
```js
if (!this.form.id.length || this.form.id.length === 13) {
        this.errors.villageId = null
      } else {
        this.errors.villageId = 'Format isian kode wilayah salah'
      }
```
Because the state when I click `batal` button is the value of `this.form.id` will be reset as an empty string, when I click again add button the field is an error. So to handle this I will refactor as code above.

Reply:
I see, okay @doohanas 👍🏼